lib/Auth/Process/Perunauthorize.php

```diff
@@ -66,6 +81,14 @@ public function __construct($config, $reserved)
             unset($config['regex']);
         }
 
+        if (isset($config[self::MESSAGE]) && is_array($config[self::MESSAGE])) {
+            $this->message = $config[self::MESSAGE];
+        }
+
+        if (isset($config[self::ADMINISTRATION_CONTACT]) && is_string($config[self::ADMINISTRATION_CONTACT])) {
+            $this->administrationContactAttribute = $config[self::ADMINISTRATION_CONTACT];
+        }
+
         foreach ($config as $attribute => $values) {
             if (is_string($values)) {
                 $values = [$values];
```

Comment:
use [Configuration](https://github.com/simplesamlphp/simplesamlphp/blob/master/lib/SimpleSAML/Configuration.php) class, e.g. `$config->getString(self::ADMINISTRATION_CONTACT, null)`

Reply:
Unfortunately, $config is not an instance of Configuration class here, it's just a simple array [](https://github.com/simplesamlphp/simplesamlphp/blob/master/lib/SimpleSAML/Auth/ProcessingFilter.php#L49)